yeonjae/programmers-1/이상한문자만들기.js

```diff
@@ -0,0 +1,19 @@
+function solution(s) {
+  const result = [];
+  const arr = s.split(" ");
+
+  for (let el of arr) {
+    let temp = "";
+
+    for (let i = 0; i < el.length; i++) {
+      if (i % 2 === 0) {
+        temp += el[i].toUpperCase();
+      } else {
+        temp += el[i].toLowerCase();
+      }
+    }
+    result.push(temp);
+  }
+
+  return result.join(" ");
+}
```

Comment:
```suggestion
function solution(s) {
  const result = [];
  const arr = s.toLowerCase().split(" ");

  for (let el of arr) {
    let temp = "";

    for (let i = 0; i < el.length; i++) {
      temp += i % 2 === 0 ? el[i].toUpperCase() : el[i];
    }
    result.push(temp);
  }
  return result.join(" ");
}
```
- 잘푸셨습니다 ㅎㅎ arr에 미리 toLowerCase()를 해서 반복문 내에서의 코드 길이를 최적화를 진행했고, if ... else 문도 좋았지만 `삼항 연산자`도 꽤나 깔끔하다고 느꼈습니다 👍
- 아래는 `map`을 이용한 풀이입니다 참고만해주세요
```js
function solution(s) {
  return s
    .toLowerCase()
    .split(" ")
    .map((word) => {
      return word
        .split("")
        .map((el, i) => (i % 2 === 0 ? el.toUpperCase() : el))
        .join("");
    })
    .join(" ");
}
```